### PR TITLE
C handle init

### PR DESCRIPTION
**Pull request by trevorhardy**
Add `__init__()` definitions for all classes that inherit from "_HelicsCHandle()". This fix was necessary for allowing the implementation of a query callback (specifically providing the init to "HelicsQueryBuffer").

### CODE REVIEW

**Comment by phlptp** [comment]
Do we know how to fix the failures here

**Comment by trevorhardy** [comment]
For all the functions that inherited from "_HelicsCHandle" I implemented the fix that Mark put together (initializing with the super classes init) and I'm 99% sure that this will provide at least a good-enough band-aid to make things not fail like they were for us. Once we have a new PyHELICS expert they should go through capi.py and figure out how to do this better.

https://github.com/GMLC-TDC/pyhelics/issues/75

**Comment by phlptp** [comment]
Getting a lot of 
```
tests/test_api.py:13: in <module>
    import helics as h
helics/__init__.py:2: in <module>
    from .capi import *
E     File "/Users/runner/work/pyhelics/pyhelics/helics/capi.py", line 1477
E       class HelicsQueryBuffer(_HelicsCHandle):
E       ^
E   SyntaxError: invalid syntax
``` 
in the testing

**Comment by trevorhardy** [comment]
Let me see if there's something I've obviously done wrong.

**Comment by trevorhardy** [comment]
Stupid missing ")", double-checking I didn't copy-paste this everywhere and then I'll push back up. Should be just a minute.

**Comment by trevorhardy** [comment]
Ok, let's try this again.

**Comment by trevorhardy** [comment]
I'm feeling good about this one....

**Comment by nightlark** [comment]
Maybe the filter test should be marked as skip on Windows? Access violation sounds like it is crashing in the C/C++ part of the HELICS library.

**Comment by trevorhardy** [comment]
I don't know enough to say whether skipping that test is a good idea but I like seeing one more green checkmark.

**Comment by nightlark** [comment]
Interesting how it passes for the CI job triggered by the "push"/commit, but fails for the "pull_request" trigger.

File: helics/capi.py [hunk 1]
```diff
@@ -9364,7 +9367,8 @@ def helicsQueryBufferFill(buffer: HelicsQueryBuffer, string: str):
     """
     f = loadSym("helicsQueryBufferFill")
     err = helicsErrorInitialize()
-    f(buffer.handle, string, len(string), err)
+    str_ptr = ffi.new('char[]', string.encode('utf-8'))
```
Comment (nightlark) [inline helics/capi.py:9370]: Is HELICS expecting a utf-8 encoded string?

Comment (trevorhardy) [reply helics/capi.py:9370]: This is what Mark Eberlein put in and it works so I guess, "yes"? It might make more sense to choose `ascii`. Let me know if you want me to change something.

Comment (nightlark) [reply helics/capi.py:9370]: It looks like the test code in HELICS is just shoving a regular `char *` into the query buffer, which seems to imply that any encoding is possible? I think this function should have an argument that specifies the encoding to use (or no encoding/raw bytes). @phlptp

File: helics/capi.py [hunk 2]
```diff
@@ -1392,11 +1392,13 @@ def message_id(self, v):
 
 
 class HelicsQuery(_HelicsCHandle):
-    pass
+    def __init__(self, handle, cleanup=False):
+        super(HelicsQuery, self).__init__(handle, cleanup)
```
Comment (kdheepak) [inline helics/capi.py:1396]: Having `pass` should be equivalent to these lines, so I'm not quite sure why this change is needed.

Comment (trevorhardy) [reply helics/capi.py:1396]: When we just had `pass` in there we were getting the error below.
```
    f(buffer.handle, string, len(string), err)
      ^^^^^^^^^^^^^
AttributeError: cdata 'void *' has no attribute 'handle'
```

Calling the super's init() solved the problem. Generally, I didn't think a class called it's super's init but you know WAY MORE about these things than I do.

Comment (trevorhardy) [reply helics/capi.py:1396]: If you're interested @kdheepak , I'm sure we can set up a call to go over this. Or, you can try running the [connector example](https://github.com/GMLC-TDC/HELICS-Examples/tree/main/user_guide_examples/advanced/advanced_connector/interface_creation) with and without the above changes to capi.py to see if you get similar behavior. You'll need to build the HELICS "dev" branch to get the connector app.

File: helics/capi.py (hunk 1 above)
Comment (trevorhardy) [reply helics/capi.py:9370]: @phlptp; any input on what you think we should do here? I don't have a philosophical problem with supporting data structures (raw bytes) as query responses if there's a need to do so **BUT** I suspect restricting this to ASCII or UTF-8 will make it easier to maintain the code and easier for users to understand what they are doing. Off the top of my head, I can't think of a use case where it would be important to support a raw bytes data structure where something like JSON wouldn't also be sufficient.

Comment (phlptp) [reply helics/capi.py:9370]: I think I would be fine restricting it in python.   In the lower levels there are a few use cases for queries/commands getting raw binary data that I want to leave open, but that would be really complicated to use in python so don't really think we would need to support it, and if someone wanted and had the ability they could directly access the library calls.

File: helics/capi.py (hunk 2 above)
Comment (trevorhardy) [reply helics/capi.py:1396]: Does Github update automatically or manually? Right now it's showing v3.4.0 as the latest release.
Screenshot 2024-02-09 at 9 59 01 AM

Comment (phlptp) [reply helics/capi.py:1396]: apparently have to actively select a check box for that now,  now 3.5 should be the latest

Comment (nightlark) [reply helics/capi.py:1396]: I'm still seeing the checkbox selected by default when I go to draft a release. It might remember the last selection that was made for a release.

I figured it just hadn't been marked as latest while the remaining interfaces were being updated.

File: helics/capi.py [hunk 3]
```diff
@@ -2823,7 +2825,8 @@ class HelicsCallbackFederate(HelicsCombinationFederate):
     pass
 
 class HelicsDataBuffer(_HelicsCHandle):
-    pass
+    def __init__(self, handle, cleanup=False):
+        super(HelicsDataBuffer, self).__init__(handle, cleanup)
```
Comment (nightlark) [inline helics/capi.py:2829]:
```suggestion
        super().__init__(handle, cleanup)
```

File: helics/capi.py [hunk 4]
```diff
@@ -1392,11 +1392,13 @@ def message_id(self, v):
...
 class HelicsQuery(_HelicsCHandle):
-    pass
+    def __init__(self, handle, cleanup=False):
+        super(HelicsQuery, self).__init__(handle, cleanup)
 
 
 class HelicsQueryBuffer(_HelicsCHandle):
-    pass
+    def __init__(self, handle, cleanup=False):
+        super(HelicsQueryBuffer, self).__init__(handle, cleanup)
```
Comment (nightlark) [inline helics/capi.py:1401]:
```suggestion
        super().__init__(handle, cleanup)
```

File: helics/capi.py (hunk 2 above)
Comment (nightlark) [inline helics/capi.py:1396]:
```suggestion
        super().__init__(handle, cleanup)
```

**Review by nightlark — COMMENTED** [review]
Use Python 3 style `super()`.

**Comment by nightlark** [comment]
We dropped Python 2 support.

File: helics/capi.py (hunk 2 above)
Comment (kdheepak) [reply helics/capi.py]: This error is telling me that `buffer` is a `cdata 'void *'` object and has no attribute 'handle'.

```
    f(buffer.handle, string, len(string), err)
      ^^^^^^^^^^^^^
AttributeError: cdata 'void *' has no attribute 'handle'
```

That's being called in:

https://github.com/GMLC-TDC/pyhelics/blob/5516142dfa81de726357c6986444c5f310240e24/helics/capi.py#L9595-L9609

You can see that `buffer` should be a `HelicsQueryBuffer` and not a `cdata 'void *'`. That's telling me that the wrong data is being passed in for some reason.

I do see the `HelicsQueryBuffer` is being created correctly here:

https://github.com/GMLC-TDC/HELICS-Examples/blob/main/user_guide_examples/advanced/advanced_connector/interface_creation/Charger.py#L169
https://github.com/GMLC-TDC/HELICS-Examples/blob/main/user_guide_examples/advanced/advanced_connector/interface_creation/Battery.py#L116

I'll try to run the example to see what is going on when I get the chance.

**Review by phlptp — APPROVED** [review]

**Review by nightlark — APPROVED** [review]
